File: client-module-server/src/main/java/br/edu/les/module/client/domain/Endereco.java
package br.edu.les.module.client.domain;

import lombok.*;
import org.springframework.stereotype.Component;

import javax.persistence.*;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Component
@Entity
@Table(name = "endereco")
public class Endereco extends EntidadeDominio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "end_id")
    private Integer id;

    @Column(name = "end_logradouro")
    private String logradouro;

    @Column(name = "end_numero")
    private String numero;

    @Column(name = "end_bairro")
    private String bairro;

    @Column(name = "end_cep")
    private String cep;

    @Column(name = "end_complemento")
    private String complemento;

    @ManyToOne()
    @JoinColumn(name = "end_cidade_id")
    private Cidade cidade;

    @ManyToOne()
    @JoinColumn(name = "end_cliente_id")
    private Cliente cliente;

    @ManyToOne()
    @JoinColumn(name = "end_status_id")
    private Status status;

    @ManyToOne()
    @JoinColumn(name = "end_ten_id")
    private TipoEndereco tipoEndereco;

}

File: client-module-server-node/src/models/Endereco.ts
import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
import { Cidade } from "./Cidade";
import { Cliente } from "./Cliente";
import { Status } from "./Status";
import { TipoEndereco } from "./TipoEndereco";

@Entity("endereco")
class Endereco {
   
    @PrimaryGeneratedColumn({name: "end_id"})
    id: number;

    @Column({name: "end_logradouro"})
    logradouro: string;

    @Column({name: "end_numero"})
    numero: string;

    @Column({name: "end_bairro"})
    bairro: string;

    @Column({name: "end_cep"})
    cep: string;

    @Column({name: "end_complemento"})
    complemento: string;

    @ManyToOne(type => Cidade, cidade => cidade.enderecos, {eager: true})
    @JoinColumn({name: "end_cidade_id"})
    cidade: Cidade;

    @ManyToOne(type => Cliente, cliente => cliente.enderecos)
    @JoinColumn({name: "end_cliente_id"})
    cliente: Cliente;

    @ManyToOne(type => Status, {eager: true})
    @JoinColumn({name: "end_status_id"})
    status: Status;

    @ManyToOne(type => TipoEndereco, {eager: true})
    @JoinColumn({name: "end_ten_id"})
    tipoEndereco: TipoEndereco;
}

export { Endereco };